Dockerfile.rhoai

```diff
@@ -0,0 +1,27 @@
+# Build the manager binary
+FROM registry.access.redhat.com/ubi9/go-toolset:1.21 as builder
+
+WORKDIR /workspace
+
+RUN git config --global --add safe.directory /workspace
+
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# Copy the go source
+COPY . .
+
+# Build
+RUN make build GO_BUILD_ENV='CGO_ENABLED=1 GOOS=linux'
+
+FROM registry.redhat.io/ubi9/ubi:latest
```

Comment:
Should it be from `registry.access.redhat.com`?

Reply:
I saw in rhods-cpass that it is using `registry.redhat.io` for base image, may I update it to `registry.access.redhat.com` ?

Reply:
With `registry.redhat.io` it requires authentication. With `registry.access.redhat.com` the ubi image was pulled without issue.

Reply:
Tested on KinD and OpenShift with `registry.access.redhat.com`, works as expected.

Reply:
I have updated registry to `registry.access.redhat.com`..
Thank you @ChristianZaccaria for testing and reviewing this PR! :)